src/shared/lib/alterationCountHelpers.ts

```diff
@@ -272,3 +295,217 @@ export function countAlterationOccurences(
         }
     );
 }
+
+export function countPatientAlterationOccurences(
```

Comment:
this is a pretty long function with nested function calls. would be nice to simplify/flatten this procedure but not sure if there is an easy way to do that.

Reply:
this function is based on the also lengthy ```countAlterationOccurences``` function but with some additional logic to account for the fact we are working with patients. i tried to combine these two into one but that proved to be very sloppy and difficult to read. i don't think it would be easy to simplify this right now but maybe can look into making both these functions cleaner in the future